pr_agent/tools/pr_extended_code_suggestions.py

```diff
@@ -0,0 +1,209 @@
+            data['Code suggestions'] = await self.rank_suggestions(data['Code suggestions'])
+
+        logging.info('Preparing PR review...')
+        if get_settings().config.publish_output:
+            logging.info('Pushing PR review...')
+            self.git_provider.remove_initial_comment()
+            logging.info('Pushing inline code comments...')
+            self.push_inline_code_suggestions(data)
+
+    async def _prepare_prediction(self, model: str) -> dict:
+        logging.info('Getting PR diff...')
+        patches_diff_list = get_pr_multi_diffs(self.git_provider, self.token_handler, model,
+                                               max_calls=get_settings().pr_extendeted_code_suggestions.max_number_of_calls)
+
+        logging.info('Getting multi AI prediction...')
+        prediction_list = []
+        for i, patches_diff in enumerate(patches_diff_list):
+            logging.info(f"Processing chunk {i + 1} of {len(patches_diff_list)}")
+            self.patches_diff = patches_diff
+            prediction = await self._get_prediction(model)
+            prediction_list.append(prediction)
+        self.prediction_list = prediction_list
+
+        data = {}
+        for prediction in prediction_list:
+            self.prediction = prediction
+            data_per_chunk = self._prepare_pr_code_suggestions()
+            if "Code suggestions" in data:
+                data["Code suggestions"].extend(data_per_chunk["Code suggestions"])
+            else:
+                data.update(data_per_chunk)
+        self.data = data
+        return data
+
+    async def _get_prediction(self, model: str):
+        variables = copy.deepcopy(self.vars)
+        variables["diff"] = self.patches_diff  # update diff
+        environment = Environment(undefined=StrictUndefined)
+        system_prompt = environment.from_string(get_settings().pr_code_suggestions_prompt.system).render(variables)
+        user_prompt = environment.from_string(get_settings().pr_code_suggestions_prompt.user).render(variables)
+        if get_settings().config.verbosity_level >= 2:
+            logging.info(f"\nSystem prompt:\n{system_prompt}")
+            logging.info(f"\nUser prompt:\n{user_prompt}")
+        response, finish_reason = await self.ai_handler.chat_completion(model=model, temperature=0.2,
+                                                                        system=system_prompt, user=user_prompt)
+
+        return response
+
+    def _prepare_pr_code_suggestions(self) -> str:
+        review = self.prediction.strip()
+        try:
+            data = json.loads(review)
+        except json.decoder.JSONDecodeError:
+            if get_settings().config.verbosity_level >= 1:
+                logging.info(f"Could not parse json response: {review}")
+            data = try_fix_json(review, code_suggestions=True)
+        return data
+
+    def push_inline_code_suggestions(self, data):
+        code_suggestions = []
```

Comment:
**Suggestion:** Consider adding a docstring to the `push_inline_code_suggestions` method to explain its purpose, parameters, and return value.
```suggestion
    def push_inline_code_suggestions(self, data):
            """
            Pushes inline code suggestions to the PR.

            Args:
                data (dict): The data containing the code suggestions.
            """
            code_suggestions = []
```